decidim-debates/lib/decidim/debates/debate_serializer.rb

```diff
@@ -37,12 +37,17 @@ def serialize
           component: { id: component.id },
           reference: debate.reference,
           comments: debate.comments_count,
-          followers: debate.follows.size,
+          followers_count: debate.followers.size,
           url:,
           last_comment_at: debate.last_comment_at,
+          last_comment_by_id: debate.last_comment_by_id,
+          last_comment_by_type: debate.last_comment_by_type,
```

Comment:
I mentioned that we should use the authors metadata instead of adding the raw information, as this isn't useful for actual users of this file. This means implementing  something like this: 

https://github.com/greenwoodt/decidim/blob/91e80713f33a2dbb96e6807844145ee19fab56dd/decidim-debates/lib/decidim/debates/debate_serializer.rb#L67

```ruby
# (...)
last_comment_by: {
            **last_comment_by_fields
},

# (...)

def last_comment_by_fields 
{
  id: debate.last_comment_by.id,
  name: author_name(debate.last_comment_by),
  url: author_url(debate.last_comment_by)
}
end
``` 

Then, when this is working with tests, I'd rename the author_name and author_url methods to user_name and user_url to make them make sense with the new approach

Reply:
Hmmm I seem to get a Nil value when trying to retrieve last_comment_by objects?